Dockerfile

```diff
@@ -0,0 +1,87 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM golang:1.10-alpine AS build
+
+RUN apk add --update \
+      curl \
+      git \
+      make \
+      py-pip \
+      python \
+      python-dev \
+      && true
+
+RUN mkdir /out
+RUN mkdir -p /out/etc/apk && cp -r /etc/apk/* /out/etc/apk/
+RUN apk add --no-cache --initdb --root /out \
+    alpine-baselayout \
+    busybox \
+    ca-certificates \
+    coreutils \
+    git \
+    libc6-compat \
+    libgcc \
+    libstdc++ \
+    python \
+    && true
+
+ENV SKAFFOLD $GOPATH/src/github.com/GoogleContainerTools/skaffold
+RUN mkdir -p "$(dirname ${SKAFFOLD})"
+COPY . $SKAFFOLD
+
+WORKDIR $SKAFFOLD
+RUN make \
+    && cp out/skaffold /out/usr/local/bin/skaffold
+
+WORKDIR /out
+
+RUN ln -s /lib /lib64
+
+ENV KUBECTL_VERSION v1.10.6
+RUN curl --silent --location "https://dl.k8s.io/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" --output usr/local/bin/kubectl \
+    && chmod +x usr/local/bin/kubectl
+
+ENV DOCKER_VERSION 18.03.0
+RUN curl --silent --location "https://download.docker.com/linux/static/stable/x86_64/docker-${DOCKER_VERSION}-ce.tgz" \
+    | tar xz docker/docker \
+    && mv docker/docker usr/local/bin/docker && rm -rf docker
+
+ENV CLOUD_SDK_VERSION 206.0.0
+RUN curl --silent --location "https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-${CLOUD_SDK_VERSION}-linux-x86_64.tar.gz" \
+    | tar xz \
+    && pip install --root=/out crcmod==1.7
+
+ENV HELM_VERSION v2.9.1
+RUN curl --silent --location "https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VERSION}-linux-amd64.tar.gz" \
+    | tar xz linux-amd64/helm \
+    && mv linux-amd64/helm usr/local/bin/helm
+
+ENV DOCKER_CREDENTIAL_GCR_VERSION 1.5.0
+RUN curl --silent --location "https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v${DOCKER_CREDENTIAL_GCR_VERSION}/docker-credential-gcr_linux_amd64-${DOCKER_CREDENTIAL_GCR_VERSION}.tar.gz" \
+    | tar xz ./docker-credential-gcr \
+    && mv docker-credential-gcr usr/local/bin/docker-credential-gcr
+# TODO: docker-credential-gcr configure-docker
+
+ENV KUSTOMIZE_VERSION 1.0.6
+RUN curl --silent --location "https://github.com/kubernetes-sigs/kustomize/releases/download/v${KUSTOMIZE_VERSION}/kustomize_${KUSTOMIZE_VERSION}_linux_amd64" --output usr/local/bin/kustomize \
+    && chmod +x usr/local/bin/kustomize
+
+ENV BAZEL_VERSION 0.16.1
+RUN curl --silent --location "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-linux-x86_64" --output usr/local/bin/bazel \
+    && chmod +x usr/local/bin/bazel
```

Comment:
Looks like it needs glibc... 😿

```
dfc0b8b9e9f8:/go# ldd /usr/local/bin/bazel
        /lib64/ld-linux-x86-64.so.2 (0x7f2384c48000)
        librt.so.1 => /lib64/ld-linux-x86-64.so.2 (0x7f2384c48000)
        libdl.so.2 => /lib64/ld-linux-x86-64.so.2 (0x7f2384c48000)
        libpthread.so.0 => /lib64/ld-linux-x86-64.so.2 (0x7f2384c48000)
        libm.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7f2384c48000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x7f23848f6000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x7f23846e4000)
        libc.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7f2384c48000)
Error relocating /usr/local/bin/bazel: __realpath_chk: symbol not found
Error relocating /usr/local/bin/bazel: __memcpy_chk: symbol not found
Error relocating /usr/local/bin/bazel: __sprintf_chk: symbol not found
dfc0b8b9e9f8:/go# bazel
Error relocating /usr/local/bin/bazel: __realpath_chk: symbol not found
Error relocating /usr/local/bin/bazel: __memcpy_chk: symbol not found
Error relocating /usr/local/bin/bazel: __sprintf_chk: symbol not found
dfc0b8b9e9f8:/go#
```

We have `libc6-compat` already, and I tried using instructions from https://github.com/winfinit/mongodb-prebuilt/issues/35 without any luck. Will have another go later on.

Reply:
Also filed https://github.com/bazelbuild/bazel/issues/5891.

Reply:
For now, I'm gonna compile bazel, as it's relatively easy to do. However, it'd be good to compare how much do we win (in terms of image size) from having alpine vs e.g. ubuntu.

Reply:
Also filed https://github.com/bazelbuild/bazel/issues/5909.